Replace unittestzero with native asserts

m8ttyB opened the pull request:
Cleanup:
* Pin packages in the the `requirements.txt` to the latest and greatest (addresses issue #70 ) .
* Update tests to use Python's native `assert`.
* Remove `xfails` that are no longer necessary.

davehunt commented on requirements.txt:
```diff
@@ -1,10 +1,5 @@
-pytest-xdist==1.8
-unittestzero
-requests==2.4.3
+py==1.4.30
```
I know we do this elsewhere, but I'd like to let the version of pytest we use determine the version of py that it requires. In my opinion we can remove this line entirely.

davehunt commented on tests/base.py:
```diff
@@ -26,7 +25,7 @@ def _head_request(self, url, user_agent=_user_agent_firefox,
             request_url = self._build_request_url(url, params)
 
-            Assert.fail('Failing URL: %s.\nError message: %s' % (request_url, e))
+            assert False, 'Failing URL: %s.\nError message: %s' % (request_url, e)
```
We should use `raise AssertionError('reason')` wherever we previously used `Assert.fail('reason')`.

davehunt commented on tests/test_redirects.py:
```diff
@@ -21,50 +21,22 @@ def test_that_checks_redirect_using_incorrect_query_values(self, base_url):
-            )
-        )
+        assert (response.status_code == requests.codes.not_found,
+                self.response_info_failure_message(base_url, param, response))
```
We've decided to be consistent with the order we have our expected vs actual when doing comparisons. Expected should be first. See our styleguide for details: https://wiki.mozilla.org/QA/Execution/Web_Testing/Docs/Automation/StyleGuide#Assertions

davehunt commented on tests/test_redirects.py:
```diff
@@ -21,50 +21,22 @@ def test_that_checks_redirect_using_incorrect_query_values(self, base_url):
-                response))
-        )
+        assert (parsed_url.scheme == 'http', 'Failed by redirected to incorrect scheme %s. \n %s' %
+                (parsed_url.scheme, self.response_info_failure_message(base_url, param, response)))
```
We shouldn't need to include the scheme in the message, as pytest will recognise we're doing a comparison and helpfully show both expected and actual values in the traceback.

davehunt commented on tests/test_redirects.py:
```diff
@@ -206,58 +122,33 @@ def test_redirect_for_firefox_aliases(self, base_url, product_alias):
             ):
-                Assert.contains('/%s/' % 'win32', parsed_url.path,
-                                '\n %s' % self.response_info(response))
+                assert '/win32/' in parsed_url.path, '\n %s' % self.response_info(response)
```
I suspect we don't need the '\n' before the message now.

davehunt commented:
Can we add a `travis.yml` with simple pep8 checking? Also, let's not squash this when we merge as the commits are nice and distinct.

m8ttyB commented:
Spot on with the review - I've addressed the comments save for the `travis.yaml` config file. At the moment the files do not meet `pep8` compliance. I will spin this off into a separate git issue.

davehunt commented on tests/test_redirects.py:
```diff
@@ -86,37 +58,18 @@ def test_that_checks_redirect_using_locales_and_os(
-    )
+        assert (requests.codes.ok == response.status_code,
+                'Redirect failed with HTTP status %s. \n %s' %
+                (response.status_code, self.response_info_failure_message(base_url, param, response)))
```
Nit: No need to include the actual status code in the message as pytest will show this for us.

davehunt commented on tests/test_redirects.py:
```diff
@@ -86,37 +58,18 @@ def test_that_checks_redirect_using_locales_and_os(
+
+        assert ('http' == parsed_url.scheme,
+               'Failed to redirect to the correct scheme. %s' %
+                (self.response_info_failure_message(base_url, param, response)))
```
Nit: No need to put the string formatting argument in brackets when there's only one.

davehunt commented on tests/test_redirects.py:
```diff
@@ -206,58 +122,33 @@ def test_redirect_for_firefox_aliases(self, base_url, product_alias):
+            assert (parsed_url.netloc in ['download.cdn.mozilla.net', 'edgecastcdn.net',
+                    'download-installer.cdn.mozilla.net', 'cloudfront.net', 'ftp.mozilla.org'],
+                    'Failed, redirected to unknown host %s. \n %s' %
+                    (parsed_url.netloc, self.response_info_failure_message(base_url, param, response)))
```
Nit: No need to put the actual value into the failure message as pytest will do this for us.

davehunt commented on tests/test_redirects.py:
```diff
@@ -206,58 +122,33 @@ def test_redirect_for_firefox_aliases(self, base_url, product_alias):
             ):
-                Assert.contains('/%s/' % 'win32', parsed_url.path,
-                                '\n %s' % self.response_info(response))
+                assert '/win32/' in parsed_url.path, '%s' % self.response_info(response)
```
Nit: `self.response_info(response)` will be coerced into a string without the need for the string formatting.

m8ttyB replied:
Nits addressed

m8ttyB replied:
Nits addressed.

m8ttyB commented:
@rbillings and @bobsilverberg - do you have a moment for an r?

davehunt commented on tests/test_redirects.py:
```diff
@@ -86,37 +58,18 @@ def test_that_checks_redirect_using_locales_and_os(
-    )
+        assert (requests.codes.ok == response.status_code,
+                'Redirect failed with HTTP status. %s' %
+                response.status_code, self.response_info_failure_message(base_url, param, response))
```
We need to remove `response.status_code` here, otherwise there are too many arguments.

m8ttyB replied:
Drat .. I think a messed up a previous rebase and reintroduced some artifacts.

davehunt commented on tests/test_redirects.py:
```diff
@@ -128,41 +81,21 @@ def test_stub_installer_redirect_for_en_us_and_win(self, base_url, product):
-        )
+        assert (requests.codes.ok == response.status_code,
+                'Redirect failed with HTTP status. %s' %
+                response.status_code, self.response_info_failure_message(base_url, param, response))
```
As above, too many arguments.

davehunt commented on tests/test_redirects.py:
```diff
@@ -206,58 +122,33 @@ def test_redirect_for_firefox_aliases(self, base_url, product_alias):
-            )
+            assert (requests.codes.ok == response.status_code,
+                    'Redirect failed with HTTP status. %s' %
+                    response.status_code, self.response_info_failure_message(base_url, param, response))
```
As above, too many arguments.

davehunt commented on requirements.txt:
```diff
@@ -1,10 +1,4 @@
-unittestzero
-requests==2.4.3
+pytest==2.8.0
+pytest-xdist==1.13.1
```
I missed this earlier (or perhaps ignored) but we've been removing pytest-xdist from our project requirements as it's not strictly a requirement but instead a productivity boost. If we remove this however, we'll be pinned to an earlier version of pytest-xdist on our CI that appears to not be compatible with pytest 2.8 so my recommendation would be to remove this line and pin pytest to 2.7.2. This will also address issue #70.

davehunt commented:
r=me with the comments addressed. Thanks @m8ttyB!

davehunt commented:
@m8ttyB pull #72 just landed, which means you'll need to rebase, but also means you should get a pass in Travis-CI once you do!

bobsilverberg commented:
LGTM @m8ttyB. r+

m8ttyB commented:
Thx @bobsilverberg, I'm assuming this is good to merge given the 2x reviews. Merging, meow.

davehunt commented:
Merging fixup commits is not ideal, let's try to rebase and squash these in future.
